core/object/object.cpp

```diff
@@ -165,6 +165,37 @@ MethodInfo MethodInfo::from_dict(const Dictionary &p_dict) {
 	return mi;
 }
 
+uint32_t MethodInfo::get_compatibility_hash() const {
+	bool has_return = (return_val.type != Variant::NIL) || (return_val.usage & PROPERTY_USAGE_NIL_IS_VARIANT);
+
+	uint32_t hash = hash_murmur3_one_32(has_return);
+	hash = hash_murmur3_one_32(arguments.size(), hash);
+
+	if (has_return) {
+		hash = hash_murmur3_one_32(return_val.type, hash);
+		if (return_val.class_name != StringName()) {
+			hash = hash_murmur3_one_32(return_val.class_name.operator String().hash(), hash);
+		}
+	}
+
+	for (const PropertyInfo &arg : arguments) {
+		hash = hash_murmur3_one_32(arg.type, hash);
+		if (arg.class_name != StringName()) {
+			hash = hash_murmur3_one_32(arg.class_name.operator String().hash(), hash);
+		}
+	}
+
+	hash = hash_murmur3_one_32(default_arguments.size(), hash);
+	for (const Variant &v : default_arguments) {
+		hash = hash_murmur3_one_32(v.hash(), hash);
+	}
+
+	hash = hash_murmur3_one_32(flags & METHOD_FLAG_CONST ? 1 : 0, hash);
```

Comment:
Using a 32-bit hash function for an effective boolean seems a bit suboptimal. `HashMap` uses `hash_fmix32` for `uint8`, but I guess that's probably not a lot better.

Reply:
The whole of the hashing here is copied from `MethodBind::get_hash()`. It isn't really necessary for virtual methods to be the same as those for non-virtual methods, but I'd prefer them to be, so I'm going to leave this as it is.

Reply:
Using the same hash method seems fine in the terms of uniformity.

Reply:
This seems like a sensible decision. Maybe a comment alluding to this would help keep it that way :)

Reply:
In my latest push, I've added a comment explaining that this functions aims to generate hashes compatible with `MethodBind::get_hash()`. Thanks!